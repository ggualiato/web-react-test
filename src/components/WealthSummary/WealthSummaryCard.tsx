import React, { FC, useState } from "react";
import Skeleton from "react-loading-skeleton";
import ThreeDots from "../../assets/three-dots.svg";
import Button from "../Common/Button";

const classNamePrefix = "my-summary";

interface WealthSummaryProps {
    total: number;
    profitability?: number;
    gain: number;
    cdi: number;
}

export const WealthSummaryCard: FC<WealthSummaryProps> = (props) => {
    const [showMore, setShowMore] = useState(false);
    const { total, profitability, cdi, gain } = props;

    return (
        <div className={`${classNamePrefix}_container`}>
            <div className={`${classNamePrefix}_header`}>
                <span>Seu resumo</span>

                <button>
                    <img src={ThreeDots} />
                </button>
            </div>

            <div className={`${classNamePrefix}_total_container`}>
                <span>Valor investido</span>
                <span>{total?.toLocaleString("pt-br", { style: "currency", currency: "BRL" }) || <Skeleton />}</span>
            </div>

            <div className={`${classNamePrefix}_summary-container`}>
                <div className="summary-item">
                    <span>Rentabilidade/mês</span>
                    {profitability?.toLocaleString("pt-br", {
                        style: "percent",
                        maximumFractionDigits: 3,
                        minimumFractionDigits: 2,
                    }) || <Skeleton width={100} />}
                </div>

                <div className="summary-item">
                    <span>CDI</span>
                    {cdi?.toLocaleString("pt-br", {
                        style: "percent",
                        maximumFractionDigits: 2,
                        minimumFractionDigits: 2,
                    }) || <Skeleton width={100} />}
                </div>

                <div className="summary-item">
                    <span>Ganho/mês</span>
                    {gain?.toLocaleString("pt-br", { style: "currency", currency: "BRL" }) || <Skeleton width={100} />}
                </div>
            </div>

            <hr></hr>

            <footer>
                <Button onClick={() => setShowMore(!showMore)}>Ver Mais</Button>
            </footer>
            {showMore && props.children}
        </div>
    );
};
