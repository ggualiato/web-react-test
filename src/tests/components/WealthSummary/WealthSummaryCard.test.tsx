import React from "react";
import { shallow } from "enzyme";
import WealthSummaryCard from "../../../components/WealthSummary/WealthSummaryCard";

test("should render Button correctly", () => {
    const wrapper = shallow(
        <WealthSummaryCard total={30000} profitability={12.0} gain={5} cdi={102}></WealthSummaryCard>
    );
    expect(wrapper).toMatchSnapshot();
});
