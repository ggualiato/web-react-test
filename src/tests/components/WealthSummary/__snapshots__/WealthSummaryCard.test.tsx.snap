// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Button correctly 1`] = `ShallowWrapper {}`;
